feature/security/gnsi/authz/tests/authz/authz1_4_test.go

```diff
@@ -285,9 +290,14 @@ func TestAuthz1(t *testing.T) {
 
 	t.Run("Authz-1.4, Test Normal Policy", func(t *testing.T) {
 		// Pre-Test Section
-		_, policyBefore := authz.Get(t, dut)
-		t.Logf("Authz Policy of the Device %s before the Rotate Trigger is %s", dut.Name(), policyBefore.PrettyPrint(t))
-		defer policyBefore.Rotate(t, dut, uint64(time.Now().Unix()), fmt.Sprintf("v0.%v", (time.Now().UnixNano())), false)
+		statusmsg, policyBefore := authz.Get(t, dut)
+		//t.Logf("Message for first authz get %s", msg)
+		if statusmsg == nil {
+			t.Logf("Expected error FAILED_PRECONDITION seen for authz Get Request.")
+		} else {
+			t.Logf("Authz Policy of the Device %s before the Rotate Trigger is %s", dut.Name(), policyBefore.PrettyPrint(t))
+			defer policyBefore.Rotate(t, dut, uint64(time.Now().Unix()), fmt.Sprintf("v0.%v", (time.Now().UnixNano())), false)
```

Comment:
why defer here?

Reply:
defer policyBefore.Rotate is not added as part of this git pull request. In this code snippet we are making sure device is Rotate ready for further tests. As we have added "FAILED PRECONDITION" validation handle we have just moved earlier code flow to else part so that if the device already has a policy before Authz test execution whether it is Rotate ready with current set of policy rules.